cli/tests/pull.bats

```diff
@@ -554,6 +554,47 @@ function add_incompatible_package() {
 
 # ---------------------------------------------------------------------------- #
 
+# bats test_tags=pull:copy:new
+@test "'pull --copy' creates path environment" {
+  make_dummy_env "owner" "name"
+
+  # dummy environment has no packages to resolve
+  export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.json"
+
+  # dummy remote as we are not actually pulling anything
+  run "$FLOX_BIN" pull --remote owner/name --copy
+  assert_success
+  assert [ ! -e ".flox/env.lock" ]
+  assert [ $(cat .flox/env.json | jq -r '.name') == "name" ]
+  assert [ $(cat .flox/env.json | jq -r '.owner') == "null" ]
+  assert_output --partial "Created path environment from owner/name"
+}
+
+
+# bats test_tags=pull:copy:convert
+@test "'pull --copy' converts managed env to path environment" {
+  make_dummy_env "owner" "name"
+
+  # dummy environment has no packages to resolve
+  export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.json"
+
+  # dummy remote as we are not actually pulling anything
+  run "$FLOX_BIN" pull --remote owner/name
+  assert_success
+  assert [ -e ".flox/env.lock" ]
+  assert [ $(cat .flox/env.json | jq -r '.name') == "name" ]
+  assert [ $(cat .flox/env.json | jq -r '.owner') == "owner" ]
+
+  run "$FLOX_BIN" pull --copy
+  assert_success
+  assert [ ! -e ".flox/env.lock" ]
+  assert [ $(cat .flox/env.json | jq -r '.name') == "name" ]
+  assert [ $(cat .flox/env.json | jq -r '.owner') == "null" ]
+  assert_output --partial "Created path environment from owner/name"
+}
+
```

Comment:
suggestion nonblocking: I don't think we care about most of the cases too much, but it might be good to add coverage for at least one of the more complicated `handle_pull_result_cases` (like a system missing or something) to make sure that's all behaving how we expect